sdk/monitor/monitor-opentelemetry-exporter/src/generated/models/index.ts

```diff
@@ -6,573 +6,387 @@
-   */
+  /** Indication of successful or unsuccessful call. */
   success?: boolean;
-  /**
-   * Collection of custom properties.
-   */
+  /** Collection of custom properties. */
   properties?: { [propertyName: string]: string };
-  /**
-   * Collection of custom measurements.
-   */
+  /** Collection of custom measurements. */
   measurements?: { [propertyName: string]: number };
 };
 
-/**
- * An instance of Request represents completion of an external request to the application to do work and contains a summary of that request execution and the results.
- */
+/** An instance of Request represents completion of an external request to the application to do work and contains a summary of that request execution and the results. */
 export type RequestData = MonitorDomain & {
-  /**
-   * Identifier of a request call instance. Used for correlation between request and other telemetry items.
-   */
+  /** Identifier of a request call instance. Used for correlation between request and other telemetry items. */
   id: string;
-  /**
-   * Name of the request. Represents code path taken to process request. Low cardinality value to allow better grouping of requests. For HTTP requests it represents the HTTP method and URL path template like 'GET /values/{id}'.
-   */
+  /** Name of the request. Represents code path taken to process request. Low cardinality value to allow better grouping of requests. For HTTP requests it represents the HTTP method and URL path template like 'GET /values/{id}'. */
   name?: string;
-  /**
-   * Request duration in format: DD.HH:MM:SS.MMMMMM. Must be less than 1000 days.
-   */
+  /** Request duration in format: DD.HH:MM:SS.MMMMMM. Must be less than 1000 days. */
   duration: string;
-  /**
-   * Indication of successful or unsuccessful call.
-   */
+  /** Indication of successful or unsuccessful call. */
   success: boolean;
-  /**
-   * Result of a request execution. HTTP status code for HTTP requests.
-   */
+  /** Result of a request execution. HTTP status code for HTTP requests. */
   responseCode: string;
-  /**
-   * Source of the request. Examples are the instrumentation key of the caller or the ip address of the caller.
-   */
+  /** Source of the request. Examples are the instrumentation key of the caller or the ip address of the caller. */
   source?: string;
-  /**
-   * Request URL with all query string parameters.
-   */
+  /** Request URL with all query string parameters. */
   url?: string;
-  /**
-   * Collection of custom properties.
-   */
+  /** Collection of custom properties. */
   properties?: { [propertyName: string]: string };
-  /**
-   * Collection of custom measurements.
-   */
+  /** Collection of custom measurements. */
   measurements?: { [propertyName: string]: number };
 };
+
+/** Known values of {@link DataPointType} that the service accepts. */
+export enum KnownDataPointType {
+  Measurement = "Measurement",
+  Aggregation = "Aggregation"
+}
```

Comment:
Please consider exposing this enum in `src/index.ts`. You do not have to do it in this PR though.

Reply:
There're a few more enums here that my comment apply to.

Reply:
All swagger generated code is only used internally and we don't expect customer to reference any of these, these enums are used to build the telemetry envelopes we send to telemetry ingestion endpoint for Azure Monitor(Breeze), customer only care about OpenTelemetry classes and interfaces, I don't believe we want to expose these in index.ts.

Reply:
You're right they're not used in the convenience layer but they're mostly there for so customer could refer to to learn the known values (as indicated in the link in the comment above the corresponding string alias below it).

Reply:
We don't have a single supported scenario where customer will need these classes and enums, I would assume links in the comments are there so we understand what are the fields in the backend, maybe swagger JSON file is not correct and that is why you are seeing something unexpected here, but we definitely don't want to expose any of these in index file.

Reply:
This is expected but your take is fair, I am fine not exposing them. I just wanted to make sure that it is indeed the case we do not want to expose them.